compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -140,6 +147,14 @@ impl<'f> PerFunctionContext<'f> {
             let references = self.find_starting_references(block);
             self.analyze_block(block, references);
         }
+
+        for (_, block) in self.blocks.iter() {
+            for (value, store_instruction) in block.last_stores.iter() {
+                if self.last_loads.get(value).is_none() {
+                    self.instructions_to_remove.insert(*store_instruction);
+                }
+            }
+        }
```

Comment:
So if we never load from an address within a function we can remove all stores to it. This isn't true for reference parameters but we can't have those currently since we always inline all brillig functions into each other. Do we want to rely on this behavior into the future? May need a check for that case if not.

Reply:
sounds good. I'll add a check.

Reply:
done